### DISCUSSION TITLE
Cannot get past permission errors...

### DISCUSSION THREAD

**Discussion by DroptheHammer** [ericaltendorf/plotman#444] [Q&A] [answered]
System Info.  Ubuntu 20.04 LTS, Ryzen 3700x, 1 OS nvme (1TB), 2 Plot NVME (2TB) , 8 HDD for Dest / Farm.

So I've got Chia installed and synced.. I've got plotman configured to write logs to (log: /home/chialogs).  But no matter what sort of CH modding I do that directory. I run into 2 issues.

1. Plotman cannot seem to write there even though my user and other programs can
2. new files created there never seem to inherit the CHMOD permissions of either 766 or 777 levels that I've tried with everything from `touch` or otherwise
3. Since plotman wont even run, I cant even `ps aux` to figure out what user even is trying to write the file
4. (sort of a question) - What sort of permissions does it (plotman) even need? just read write? or read write execute?

Error Message after `plotman interactive` attached

```python-traceback
(venv) backhoe@farm1:~/chia-blockchain$ plotman interactive
Traceback (most recent call last):
  File "/home/backhoe/chia-blockchain/venv/lib/python3.8/site-packages/plotman/manager.py", line 135, in maybe_start_new_plot
    open_log_file = open(logfile, 'x')
FileNotFoundError: [Errno 2] No such file or directory: '/home/chialogs/2021-05-14T11_50_19.626392-07_00.log'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/backhoe/chia-blockchain/venv/bin/plotman", line 8, in <module>
    sys.exit(main())
  File "/home/backhoe/chia-blockchain/venv/lib/python3.8/site-packages/plotman/plotman.py", line 173, in main
    interactive.run_interactive()
  File "/home/backhoe/chia-blockchain/venv/lib/python3.8/site-packages/plotman/interactive.py", line 334, in run_interactive
    curses.wrapper(curses_main)
  File "/usr/lib/python3.8/curses/__init__.py", line 105, in wrapper
    return func(stdscr, *args, **kwds)
  File "/home/backhoe/chia-blockchain/venv/lib/python3.8/site-packages/plotman/interactive.py", line 117, in curses_main
    (started, msg) = manager.maybe_start_new_plot(
  File "/home/backhoe/chia-blockchain/venv/lib/python3.8/site-packages/plotman/manager.py", line 153, in maybe_start_new_plot
    raise Exception(message) from e
Exception: Unable to open log file.  Verify that the directory exists and has proper write permissions: '/home/chialogs/2021-05-14T11_50_19.626392-07_00.log'
```

**Comment by altendky** [comment]
That is an odd place to add a random directory.  `/home` is for user's home directories.  Anyways, plotman would be running as the user that you launched it with.  It doesn't do any setuid or anything.  Please share `ls -la /home/chialogs`.  Please copy and paste the command instead of retyping it.  The user running plotman would need...  Well, write access to the directory and perhaps execute.  I forget offhand for the latter.

**Comment by DroptheHammer** [comment]
Inside you can see some of the files I was messing around with.  right now the directory is set to 777 via CHMOD but the "newfile" I just made is still showing 664.

```backhoe@farm1:~/chialogs$ ls -la
total 8
drwxrwxrwx  2 backhoe backhoe 4096 May 15 06:01 .
drwxr-xr-x 26 backhoe backhoe 4096 May 15 06:01 ..
-rw-rw-r--  1 backhoe backhoe    0 May 14 11:36 3
-rw-rw-r--  1 backhoe backhoe    0 May 14 11:36 file
-rwxrwxrwx  1 backhoe backhoe    0 May 14 11:27 file1
-rwxrwxrwx  1 backhoe backhoe    0 May 14 11:35 file2
-rw-rw-r--  1 backhoe backhoe    0 May 15 06:01 newfile
```

**Reply by altendky** [reply]
The [umask](https://en.wikipedia.org/wiki/Umask) controls the permissions of new files.  The permissions of the directory are not inherited by files created inside it.  But, that's all sort of an aside.

You have made `/home` be owned by the `backhoe` user as well?  That should be owned by `root` generally.  Though again, I don't know of any reason that should keep plotman from writing the file.

**Comment by DroptheHammer** [comment, marked as answer]
I found out what was wrong...  Or rather.. what worked better. I needed to make the string in plotman.yaml to the following, and then this same directory worked straight away with 764 permissions.

/home/backhoe/chialogs

**Reply by altendky** [reply]
I'm glad you got something working.  I mentioned that creating directories in `/home` is not normal, but I don't know why it would be a cause of this actual problem.
